zc_events/client.py

```diff
@@ -238,11 +240,40 @@ def send_email(self, *args, **kwargs):
             with attachments {} and files {}'''
             logger.info(msg.format(email_uuid, to, from_email, attachments, files))
 
-        event_data = generate_email_data(*args, **kwargs)
+        event_data = generate_email_data(email_uuid, *args, **kwargs)
 
         if logger:
             logger.info('MICROSERVICE_SEND_EMAIL: Sent email with UUID {} and data {}'.format(
                 email_uuid, event_data
             ))
 
         self.emit_microservice_event('send_email', **event_data)
+
+    def emit_index_rebuild_event(self, event_name, resource_type, model, batch_size, serializer, queryset=None):
+        """
+        A special helper method to emit events related to index_rebuilding.
+        Note: AWS_INDEXER_BUCKET_NAME must be present in your settings.
+        """
+
+        if queryset is None:
+            queryset = model.objects.all()
+
+        objects_count = queryset.count()
+        total_events_count = int(math.ceil(objects_count / batch_size))
+        emitted_events_count = 0
+
+        while emitted_events_count < total_events_count:
+            start_index = emitted_events_count * batch_size
```

Comment:
The logic here is assuming the primary key is an incremental integer id and there are no gaps in between. This is probably not the case for most tables. The logic should change so it could support all primary key formats with any kind of order.